Packages/Keystore/Sources/LocalKeystore.swift

```diff
@@ -3,99 +3,44 @@ import Primitives
 import Store
 import WalletCore
 import Preferences
+import WalletSessionService
 
-@Observable
-public final class LocalKeystore: Keystore {
+// TODO: - Add keystore Config with direcotry, folder, etc and inject into LocalKeystore
+// TODO: - move walletStore logic on top of ManageWalletService(should be extended ) and remove it dependecy
+
+public struct LocalKeystore: Keystore {
     public let directory: URL
 
     private let documentsDirectory = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
     private let walletStore: WalletStore
-    private let preferences: Preferences
     private let walletKeyStore: WalletKeyStore
-    
-    let keystorePassword: KeystorePassword
-    
-    public var currentWalletId: Primitives.WalletId? = .none
-    public var currentWallet: Primitives.Wallet? = .none
-    
+    private let keystorePassword: KeystorePassword
+    private let walletSessionService: any WalletSessionManageable
+
     public init(
         folder: String = "keystore",
         walletStore: WalletStore,
-        preferences: Preferences,
+        preferences: ObservablePreferences,
         keystorePassword: KeystorePassword = LocalKeystorePassword()
     ) {
         let directory = URL(fileURLWithPath: String(format: "%@/%@", documentsDirectory, folder))
         self.directory = directory
         self.walletKeyStore = WalletKeyStore(directory: directory)
-        self.preferences = preferences
         self.walletStore = walletStore
         self.keystorePassword = keystorePassword
-        self.currentWalletId = switch preferences.currentWalletId {
-        case .some(let value): WalletId(id: value)
-        case .none: .none
-        }
-        self.currentWallet = getWalletById(id: preferences.currentWalletId ?? "")
+        self.walletSessionService = WalletSessionService(walletStore: walletStore, preferences: preferences)
     }
-    
+
     public var wallets: [Primitives.Wallet] {
         do {
-            return try walletStore.getWallets()
+            return try walletSessionService.getWallets()
```

Comment:
Throws error not handled, mb better use `try?`

```suggestion
            try? walletSessionService.getWallets() ?? []
```

or

```suggestion
            try? walletSessionService.getWallets().or([])
```